### PR TITLE
New prop trapFocus added to Drop and DropButton

### LINKED ISSUE #2660: Target element of Drop and DropButton get tabindex of -1
### Expected Behavior
The target element should keep a tab index so that it is accessible by keyboard interaction.

### Actual Behavior
The target element, or in the case of the DropButton, the button gets a tabindex of -1 when the drop is in an open state. 

### URL, screen shot, or Codepen exhibiting the issue
Here's an example of trying to build a reusable button with a tooltip on hover/focus using DropButton:

https://codesandbox.io/s/jv3nwx1jrv

User should be able to tab between the buttons.

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by IanKBovard**
#### What does this PR do?
Adds a new prop called trapFocus to Drop and DropButton. 
Optional prop allows users to specify if they want focus to be trapped when the drop opens.
Accepts true or false, defaults to true(to preserve original functionality).

#### Where should the reviewer start?
Drop.js, DropContainer.js, DropButton.js, FocusedContainer.js

#### What testing has been done on this PR?
Tested locally in storybook using this.
https://codesandbox.io/s/strange-williams-f7vmd?file=/src/App.js

#### How should this be manually tested?
Same as above. Should preserve original drop functionality.

#### Any background context you want to provide?
The FocusContainer that wrapped the Drop component trapped focus whenever it was open. This prevented tabbing through other buttons.
Kapture 2020-08-05 at 15 14 28

This functionality makes sense with DropButton, when you use Drop with some inputs.
Kapture 2020-08-05 at 15 17 09

#### What are the relevant issues?
closes #2660 
#### Screenshots (if appropriate)
trapFocus set to false

Kapture 2020-08-05 at 14 30 24

Kapture 2020-08-05 at 14 27 50

#### Do the grommet docs need to be updated?
Yes, will update in this pr if trapFocus is accepted as a prop name.

#### Should this PR be mentioned in the release notes?
Yes
#### Is this change backwards compatible or is it a breaking change?
Default value for trapFocus is true to keep backwards compatibility.

### CODE REVIEW

**Comment by ShimiSun** [comment]
@srallen, since you have the most background on the addressed issue, will appreciate your feedback on this PR approach.

**Comment by srallen** [comment]
@ShimiSun I think this will enable me to build a series of buttons with tooltips now that use `Drop`. The only thing I'd recommend is updating documentation for this prop and adding a story demonstrating this and why you'd use it. I believe there's an existing story showing how to build a tooltip using `Drop`, so perhaps another showing a series of components with tooltips?

**Comment by IanKBovard** [comment]
added documentation, and modified the current tooltip story

File: src/js/components/DropButton/DropButton.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ const DropButton = forwardRef(
       onClick,
       onClose,
       onOpen,
+      trapFocus = true,
```
Comment (ericsoderberghp) [inline src/js/components/DropButton/DropButton.js:23]: Do we need this? Since Drop already defaults `trapFocus` to `true`, can we let `dropProps` take care of overriding it if needed?

Comment (IanKBovard) [reply src/js/components/DropButton/DropButton.js:23]: yeah, i think you're right. just tested without it, works the same.

File: src/js/components/Drop/Drop.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ const Drop = forwardRef(
     {
       restrictFocus,
       target: dropTarget, // avoid DOM leakage
+      trapFocus = true,
```
Comment (taysea) [inline src/js/components/Drop/Drop.js:15]: Does this have the same intention of `modal` prop in Layer? if so, would we want to match the terms. If not, ignore this comment :)

https://v2.grommet.io/layer#modal

Comment (IanKBovard) [reply src/js/components/Drop/Drop.js:15]: I wouldn't say the intention is the same. Modal prevents all interaction below the layer. This just locks keyboard interaction between the modal and the window. The mouse can still interact with the page when a drop is rendered.

Comment (taysea) [reply src/js/components/Drop/Drop.js:15]: Gotcha, thanks for clarifying!

**Comment by ShimiSun** [comment]
Did you notice this change in Chromatic @IanKBovard https://www.chromatic.com/snapshot?appId=5d9774839a6eff00203f5cbf&id=5f3590dec1c8d00022d1b01e?

**Comment by IanKBovard** [comment]
> Did you notice this change in Chromatic @IanKBovard https://www.chromatic.com/snapshot?appId=5d9774839a6eff00203f5cbf&id=5f3590dec1c8d00022d1b01e?

Yes, i changed the tooltip story to cover the changes in this pr.

File: src/js/components/Drop/stories/Tooltip.js [hunk 3]
```diff
@@ -1,40 +1,188 @@
...
+            <Text color="white">{name}</Text>
+          </Box>
+        </Drop>
+      )}
+    </Box>
+  );
+};
+const Sidebar = () => {
+  return (
+    <Box direction="row" fill>
```
Comment (ShimiSun) [inline src/js/components/Drop/stories/Tooltip.js:140]: If it's a Sidebar it should use the Sidebar component.

File: src/js/components/Drop/stories/Tooltip.js [hunk 4]
```diff
@@ -1,40 +1,188 @@
...
+const Tooltips = () => {
+  return (
+    <Grommet theme={grommet} full>
+      <Sidebar />
     </Grommet>
   );
 };
 
-storiesOf('Drop', module).add('Tooltip', () => <TooltipDrop />);
+storiesOf('Drop', module).add('Tooltips', () => <Tooltips />);
```
Comment (ShimiSun) [inline src/js/components/Drop/stories/Tooltip.js:188]: I think we still need a single Tooltip story so the users can find an easy way to come up with tooltips. So separating Tooltips to a separate story would be a good approach here, unless we can cover it with a unit test and then we won't need the story, thoughts?

File: src/js/components/Drop/stories/Tooltip.js [hunk 5]
```diff
@@ -1,40 +1,188 @@
...
+                  icon={icon}
+                  index={index}
+                  name={name}
+                />
+              );
+            })}
+          </Nav>
+        </Box>
+        <Box>
+          <SidebarButton icon={<Action />} index="Action" name="Action" />
```
Comment (ShimiSun) [inline src/js/components/Drop/stories/Tooltip.js:173]: The last icon on theSidebar is looking funky
image

File: src/js/components/Drop/stories/Tooltip.js [hunk 6]
```diff
@@ -1,40 +1,188 @@
...
+  );
+};
+const Sidebar = () => {
+  return (
+    <Box direction="row" fill>
+      <Box
+        direction="column"
+        basis="15%"
+        justify="between"
+        style={{ border: 'blue solid 4px' }}
```
Comment (ShimiSun) [inline src/js/components/Drop/stories/Tooltip.js:145]: We shouldn't use `style` in the story examples, why is that needed?

File: src/js/components/Drop/stories/Tooltip.js (hunk 4 above)
Comment (IanKBovard) [reply src/js/components/Drop/stories/Tooltip.js:188]: i gotcha, makes sense. ill have a new, simpler example up soon.

**Comment by IanKBovard** [comment]
updated the storybook example
